### PR TITLE
add bindings/templating to http notifier

### PR DESCRIPTION
**Pull request by Aric1088**

### CODE REVIEW

File: http/main.go [hunk 1]
```diff
@@ -62,26 +73,40 @@ func (h *httpNotifier) SendNotification(ctx context.Context, build *cbpb.Build)
...
+		return err
+	}
+	err = json.NewEncoder(payload).Encode(buf)
 	if err != nil {
-		return fmt.Errorf("failed to marshal Build proto to JSON: %w", err)
+		return fmt.Errorf("failed to encode payload: %w", err)
 	}
-
-	req, err := http.NewRequestWithContext(ctx, http.MethodPost, h.url, bytes.NewReader(jb))
+	log.Infof("request body: %v", buf.String())
```
Comment (prabenzom) [inline http/main.go:100]: why do we want to print this out? remove?

File: http/main.go [hunk 2]
```diff
@@ -62,26 +73,40 @@ func (h *httpNotifier) SendNotification(ctx context.Context, build *cbpb.Build)
...
+	log.Infof("encoded json body: %v", payload.String())
```
Comment (prabenzom) [inline http/main.go:101]: same

File: http/main.go [hunk 3]
```diff
@@ -62,26 +73,40 @@ func (h *httpNotifier) SendNotification(ctx context.Context, build *cbpb.Build)
...
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, h.url, strings.NewReader(buf.String()))
 	if err != nil {
 		return fmt.Errorf("failed to create a new HTTP request: %w", err)
 	}
-
 	req.Header.Set("Content-Type", "application/json")
 	req.Header.Set("User-Agent", "GCB-Notifier/0.1 (http)")
-
+	yes, _ := httputil.DumpRequestOut(req, true)
```
Comment (prabenzom) [inline http/main.go:108]: remove?

**Review by prabenzom — CHANGES_REQUESTED** [review]

File: http/main.go (hunk 1 above)
Comment (Aric1088) [reply http/main.go]: Removed

File: http/main.go (hunk 3 above)
Comment (Aric1088) [reply http/main.go]: Removed

**Review by prabenzom — APPROVED** [review]

**Pull request merged**